### DISCUSSION TITLE
Implementation wasmer with ML models on IoT devices

### DISCUSSION THREAD

**Discussion by TAOFOR4** [wasmerio/wasmer#4485] [Ideas]
### Summary

Hi, I am doing a research about implement wasm with ML models like TensorFlow or so on IoT devices. I found wasi-nn but it seems only for Intel processor at current stage. Then I read the post https://wasmer.io/posts/wasmer-enables-portable-ai. So I am wondering if it is possible to run ML models through wasmer on IoT devices right now. Thank you!

**Comment by TAOFOR4** [comment]
> [RUN-106 Implementation wasmer with ML models on IoT devices](https://linear.app/wasmer/issue/RUN-106/implementation-wasmer-with-ml-models-on-iot-devices)

It says I don't have access to the workplace

**Comment by theduke** [comment]
Hey, so the AI + Webassembly landscape is still evolving, right now there are no great solutions.

You can compile code with SIMD optimizations to get CPU-only inference, but for GPU support new APIs will need to emerge, both on the Webassembly host and on the client.

`webgpu` is a promising direction with quite some activity.

We are definitely interested in pushing this forward, but it's not the current immediate focus.

**Reply by TAOFOR4** [reply]
Hi, thx for the quick reply. It helps a lot. Do you know if it is possible to run WASM on any IoT devices, like Raspberry Pi or something similar? If so, can you give me the link for that? I tried to find something relevant but failed.

Thank you!
